File: tutopy_pkg/tutopy_pkg/subscriber_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from example_interfaces.msg import String


class SubscriberNode(Node):
    def __init__(self):
        super().__init__("subscriber_node")
        self.subsciber_ = self.create_subscription(
            String, "first_publisher", self.callback_publisher, 10
        )
        self.get_logger().info("Subscriber Node has been started")

    def callback_publisher(self, msg):
        self.get_logger().info(msg.data)


def main(args=None):
    rclpy.init(args=args)

    node = SubscriberNode()

    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: tutocpp_pkg/src/subscriber_node.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/int16.hpp"

class SubscriberNode : public rclcpp::Node
{
public:
    SubscriberNode() : Node("subscriberCpp")
    {
        declare_parameter("topic_name", "first_publisher");
        get_parameter("topic_name", topic_name);
        subscriber_ = create_subscription<example_interfaces::msg::Int16>(topic_name, 10,
                                                                          std::bind(&SubscriberNode::publisherCallBack,
                                                                                    this, std::placeholders::_1));
        RCLCPP_INFO(get_logger(), "Subscriber cpp Node has been started");
    }

private:
    rclcpp::Subscription<example_interfaces::msg::Int16>::SharedPtr subscriber_;
    std::string topic_name;

    void publisherCallBack(const example_interfaces::msg::Int16::SharedPtr msg)
    {
        RCLCPP_INFO(get_logger(), "%d", msg->data);
    }
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);

    auto node = std::make_shared<SubscriberNode>();

    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}